src/frontend/next/src/components/DeleteFeedDialogButton.tsx

```diff
@@ -0,0 +1,111 @@
+import { useState, useRef } from 'react';
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogActions,
+  DialogContentText,
+  DialogTitle,
+  IconButton,
+} from '@material-ui/core';
+import { Delete } from '@material-ui/icons';
+import { makeStyles } from '@material-ui/core/styles';
+import { Feed } from '../interfaces';
+import useSiteMetadata from '../hooks/use-site-metadata';
+
+type DeleteFeedDialogButtonProps = {
+  feed: Feed;
+  deletionCallback: (id: number) => void;
+};
+
+const useStyles = makeStyles(() => ({
+  button: {
+    padding: '3px 0 3px 0',
+  },
+}));
+
+const DeleteFeedDialogButton = ({ feed, deletionCallback }: DeleteFeedDialogButtonProps) => {
+  const { id, url } = feed;
+  const classes = useStyles();
+  const { telescopeUrl } = useSiteMetadata();
+  const [open, setOpen] = useState<boolean>(false);
```

Comment:
For inferred types like this, it's not necessary to include the generic `<boolean>`.  Not wrong to do this, but not necessary.